src/screens/Exchange/CoinifyWidget.js

```diff
@@ -206,35 +227,75 @@ export default function CoinifyWidget({
         scalesPageToFitmediaPlaybackRequiresUserAction
         automaticallyAdjustContentInsets={false}
         scrollEnabled={true}
-        style={{
-          flex: 0,
-          width: "100%",
-          height: "100%",
-        }}
+        style={styles.webview}
       />
-      {isWaitingDeviceJob ? (
-        <DeviceJob
-          deviceModelId="nanoX" // NB: EditDeviceName feature is only available on NanoX over BLE.
-          meta={meta}
-          onCancel={() => {
-            settleTrade("rejected");
-            setWaitingDeviceJob(false);
-          }}
-          onDone={() => {
-            settleTrade("accepted");
-            setWaitingDeviceJob(false);
-          }}
-          steps={[
-            connectingStep,
-            accountApp(mainAccount),
-            verifyAddressOnDeviceStep(mainAccount),
-          ]}
-        />
-      ) : null}
+      <BottomModal id="DeviceActionModal" isOpened={isOpen}>
+        <View style={styles.modalContainer}>
+          {isWaitingDeviceJob ? (
+            <DeviceAction
+              action={action}
+              device={device}
+              request={{ account: mainAccount, tokenCurrency }}
+              onResult={onResult}
+            />
+          ) : (
+            <VerifyAddress
+              account={mainAccount}
+              device={device}
+              onResult={onVerify}
+            />
+          )}
+        </View>
+      </BottomModal>
     </View>
   );
 }
 
+function VerifyAddress({
+  account,
+  device,
+  onResult,
+}: {
+  account: Account,
+  device: Device,
+  onResult: (confirmed: boolean, error?: Error) => void,
+}) {
+  const { t } = useTranslation();
+
+  const onConfirmAddress = useCallback(async () => {
+    try {
+      const { address } = await getAccountBridge(account)
+        .receive(account, {
+          deviceId: device.deviceId,
+          verify: true,
+        })
+        .toPromise();
+      if (address !== account.freshAddress) {
+        throw new WrongDeviceForAccount(
+          `WrongDeviceForAccount ${account.name}`,
+          {
+            accountName: account.name,
+          },
+        );
+      }
```

Comment:
```suggestion
```

Reply:
its all on live-common and you just need to call receive (it will throw in the observable in that case of wrong device account pair) ( https://github.com/LedgerHQ/ledger-live-common/blob/269971b52b06f5c6e44bb9df9fb93e51f52d6578/src/bridge/jsHelpers.js#L343 )